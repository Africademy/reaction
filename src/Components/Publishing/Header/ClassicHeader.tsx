import { garamond } from "Assets/Fonts"
import React from "react"
import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { AuthorDateClassic } from "./AuthorDateClassic"

interface ClassicHeaderProps {
  article?: any
  date?: string
  title: any
  leadParagraph?: any
  isMobile?: any
}

export const ClassicHeader: React.SFC<ClassicHeaderProps> = props => {
  const { article, date, leadParagraph, title } = props
  return (
    <ClassicHeaderContainer>
      <Title>{title}</Title>
      {leadParagraph}
      <AuthorDateClassic
        authors={article.contributing_authors}
        author={article.author}
        date={date ? date : article.published_at}
      />
    </ClassicHeaderContainer>
  )
}

const ClassicHeaderContainer = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 900px;
  width: 100%;
  margin: 40px auto;
  box-sizing: border-box;
  text-align: center;
  p,
  > p {
    line-height: 1.35em;
    text-align: left;
    max-width: 580px;
    width: 100%;
    margin: 0 auto;
    padding-bottom: 30px;
    ${garamond("s19")};
  }

  ${pMedia.xl`padding: 0 20px;`} ${pMedia.xs`
    text-align: left;
    p, > p {
      line-height: 1.35em;
      ${garamond("s17")}
    }
  `};
`
const Title = styled.div`
  margin-bottom: 30px;
  ${garamond("s37")} ${pMedia.xs`
    ${garamond("s34")}
  `};
`
