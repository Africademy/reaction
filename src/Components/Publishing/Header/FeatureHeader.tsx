import { get } from 'lodash'
import React from "react"
import styled from "styled-components"
import { resize } from "../../../Utils/resizer"
import { pMedia } from "../../Helpers"
import { Byline } from "../Byline/Byline"
import { Fonts } from "../Fonts"

const IMAGE_QUALITY: number = 60

interface FeatureHeaderProps {
  article?: any
  vertical?: any
  title: any
  deck?: any
  image?: any
  height?: string
  size?: {
    width: number
  }
}

const FeatureHeaderComponent: React.SFC<FeatureHeaderProps> = props => {
  const { article, vertical, title, deck, image, size, height } = props
  const hero = article.hero_section
  const url = get(hero, "url", "")
  const type = get(hero, "type", "text")
  const isMobile = size.width < 600

  // Layouts
  const isFullScreenLayout = type === 'fullscreen'
  const isDesktopSplitLayout = type === 'split' && !isMobile
  const isMobileSplitLayout = type === "split" && isMobile
  const isTextLayout = type === 'text'

  return (
    <FeatureHeaderContainer data-type={type} height={height}>
      {isFullScreenLayout &&
        <div>
          {renderAsset(type)(url, article.title, image)}
          <Overlay />
        </div>}

      {isDesktopSplitLayout &&
        renderAsset(type)(url, article.title, image)}

      <HeaderTextContainer>
        <HeaderText>
          <Vertical>
            {vertical}
          </Vertical>

          <Title>
            {title}
          </Title>

          {isMobileSplitLayout &&
            renderAsset(type)(url, article.title, image)}

          <SubHeader>
            {deck &&
              <Deck>
                {deck}
              </Deck>}

            <Byline
              article={article}
              layout={type}
            />
          </SubHeader>
        </HeaderText>

        {isTextLayout &&
          renderAsset(type)(url, article.title, image)}

      </HeaderTextContainer>
    </FeatureHeaderContainer>
  )
}

FeatureHeaderComponent.defaultProps = {
  height: "100vh",
  size: {
    width: 500,
  },
}

// Helpers

function renderAsset(type: string): Function {
  const isTextType = type === 'text'

  let Layout
  let Player
  let ImageW

  if (isTextType) {
    Layout = TextAsset
    Player = Video
    ImageW = Image
  } else {
    Layout = FeatureVideoContainer
    Player = FeatureVideo
    ImageW = FeatureImage
  }

  return (url: string, title: string, image: any) => {
    const isVideo = url.includes("mp4")

    if (isVideo) {
      return (
        <Layout>
          {image}

          <Player autoPlay loop muted playsInline
            src={url}
            controls={false}
          />
        </Layout>
      )
    } else {
      const hasImage = url.length

      if (hasImage) {
        const src = resize(url, {
          width: 1600,
          quality: IMAGE_QUALITY
        })

        if (isTextType) {
          return (
            <TextAsset>
              {image}

              <ImageW
                src={src}
                alt={title}
              />
            </TextAsset>
          )
        } else {
          return (
            <ImageW src={src} alt={title}>
              {image}
            </ImageW>
          )
        }
      }
    }
  }
}

// Styles

const Div = styled.div`
  width: 100%;
  height: 100%;
  box-sizing: border-box;
`
const Overlay = Div.extend`
  position: absolute;

  // TODO: This seems to visually exacerbate the load time; do we need?
  // background-image: linear-gradient(to bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.3));
`
const Vertical = styled.div`
  ${Fonts.unica("s16", "medium")}
  margin-bottom: 10px;
  ${pMedia.sm`
    ${Fonts.unica("s14", "medium")}
  `}
`
const HeaderTextContainer = Div.extend`
  margin: auto;
`
const HeaderText = Div.extend`
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 20px;
  color: #000;
  justify-content: flex-start;
`
const FeatureImage = Div.extend`
  position: absolute;
  background-image: url(${props => (props.src ? props.src : "")});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  right: 0;
  width: 100%;
`
const FeatureVideo = styled.video`
  width: 100%;
  height: 100%;
  object-fit: cover;
`
const FeatureVideoContainer = Div.extend`
  width: 100%;
  height: 100%;
  right: 0;
  position: absolute;
  overflow: hidden;
`
const Image = styled.img`
  width: 100%;
  height: auto;
  box-sizing: border-box;
`
const Video = styled.video`
  width: 100%;
`
const TextAsset = styled.div`
  width: 100%;
  padding: 20px;
  box-sizing: border-box;
`
const SubHeader = styled.div`
  ${Fonts.unica("s19", "medium")}
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  flex-direction: row;
  ${pMedia.sm`
    align-items: flex-start;
    flex-direction: column;
  `}
`
const Title = styled.div`
  ${Fonts.unica("s100")}
  margin-bottom: 75px;
  letter-spaceing: -0.035em;
  ${pMedia.xl`
    ${Fonts.unica("s80")}
  `}
  ${pMedia.md`
    ${Fonts.unica("s65")}
  `}
  ${pMedia.xs`
    ${Fonts.unica("s45")}
  `}
`
const Deck = styled.div`
  max-width: 460px;
  margin-right: 30px;
  ${Fonts.unica("s16", "medium")}
  line-height: 1.4em;
  ${pMedia.sm`
    margin-bottom: 28px;
    ${Fonts.unica("s14", "medium")}
  `}
`
const FeatureHeaderContainer = Div.extend`
  width: 100%;
  height: ${props => props.height};
  &[data-type="text"] {
    height: auto;
    ${Title} {
      margin-bottom: 150px;
    }
  }
  &[data-type="split"] {
    ${Title} {
      flex-grow: 1;
    }
    ${HeaderText} {
      width: 50%;
    }
    ${FeatureImage} {
      width: 50%;
      border: 20px solid white;
    }
    ${FeatureVideoContainer} {
      width: 50%;
      border: 20px solid white;
    }
    ${FeatureVideo} {
      width: 50vw;
    }
    ${SubHeader} {
      align-items: flex-start;
      flex-direction: column;
    }
    ${Deck} {
      margin-bottom: 30px;
    }
    ${pMedia.xs`
      ${Title} {
        margin-bottom: 20px;
      }
      ${HeaderText} {
        width: 100%;
      }
      ${FeatureImage} {
        width: 100%;
        position: relative;
        border: 0px;
        margin-bottom: 30px;
      }
      ${FeatureVideoContainer} {
        width: 100%;
        position: relative;
        border: 0px;
        margin-bottom: 30px;
      }
      ${FeatureVideo} {
        width: 100%;
      }
    `}
  }
  &[data-type="fullscreen"] {
    ${HeaderText} {
      padding: 50px;
      color: #fff;
      justify-content: flex-end;
      margin: auto;
      text-shadow: 0 0 40px rgba(0, 0, 0, 0.4);
    }
    ${pMedia.xs`
      ${HeaderText} {
        padding: 20px;
      }
    `}
  }
`

// TODO: Fix this naming ref
export const FeatureHeader = FeatureHeaderComponent
