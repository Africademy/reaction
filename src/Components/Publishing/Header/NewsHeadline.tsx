import React from "react"
import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { Fonts } from "../Fonts"

interface NewsHeadlineProps {
  title: string
  article?: any
}

export const NewsHeadline: React.SFC<NewsHeadlineProps> = props => {
  const { title } = props
  return (
    <NewsHeadlineParent>
      <NewsHeadlineContainer>
        <Title>{title}</Title>
      </NewsHeadlineContainer>
    </NewsHeadlineParent>
  )
}

const NewsHeadlineParent = styled.div`
  margin: 0 40px;
  ${pMedia.sm`
    margin: 0 20px;
  `};
`

const NewsHeadlineContainer = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 780px;
  width: 100%;
  margin: 40px auto;
  box-sizing: border-box;
  margin-bottom: 30px;
`

const Title = styled.div`
  ${Fonts.garamond("s34")};
  font-weight: 600;
  ${pMedia.sm`
    ${Fonts.garamond("s23")}
    line-height: 1.1;
  `};
`
