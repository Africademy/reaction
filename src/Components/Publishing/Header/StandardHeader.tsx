import { garamond, unica } from "Assets/Fonts"
import React from "react"
import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { Byline } from "../Byline/Byline"

interface StandardHeaderProps {
  article?: any
  date?: string
  title: any
  vertical?: any
}

export const StandardHeader: React.SFC<StandardHeaderProps> = props => {
  const { article, date, title, vertical } = props
  return (
    <StandardHeaderParent>
      <StandardHeaderContainer>
        <Vertical>{vertical}</Vertical>
        <Title>{title}</Title>
        <Byline article={article} layout="standard" date={date && date} />
      </StandardHeaderContainer>
    </StandardHeaderParent>
  )
}

const StandardHeaderParent = styled.div`
  margin: 0 40px;
  ${pMedia.sm`
    margin: 0 20px;
  `};
`

const StandardHeaderContainer = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 1250px;
  width: 100%;
  margin: 40px auto;
  box-sizing: border-box;
  ${pMedia.sm`
    margin: 30px auto;
  `};
`
const Title = styled.div`
  ${garamond("s50")} margin-bottom: 50px;
  ${pMedia.sm`
    ${garamond("s34")}
  `};
`
const Vertical = styled.div`
  ${unica("s16", "medium")} margin-bottom: 10px;
  ${pMedia.sm`
    ${unica("s14", "medium")}
  `};
`
