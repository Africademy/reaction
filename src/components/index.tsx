import Artwork from "./artwork"
import ArtworkFilter from "./artwork_filter"
import Buttons from "./buttons/index"
import Modal from "./modal/modal"

import ArtworkGrid from "./artwork_grid"
import BorderedPulldown from "./bordered_pulldown"
import Checkbox from "./checkbox"
import Grid from "./grid"
import Icon from "./icon"
import Input from "./input"
import Nav from "./nav"
import Text from "./text"
import TextArea from "./text_area"
import TextLink from "./text_link"
import Title from "./title"

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
