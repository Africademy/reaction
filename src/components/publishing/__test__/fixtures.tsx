import { Props as ImageSetPreviewProps } from "../imageset_preview"

export const Artworks = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
    },
    artist: {
      name: "Fernando Botero",
    },
    width: 1152,
    height: 826,
  },
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
      {
        name: "Frida Kahlo",
        slug: "frida-kahlo",
      },
    ],
    width: 1152,
    height: 826,
  },
]

export const Articles = [
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "classic",
    vertical: null,
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "standard",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "feature",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
]

export const Authors = [
  {
    image_url: "https://artsy-media-uploads.s3.amazonaws.com/9vcX6FR21rKHatmvJ8K0sg%2FAbigail.jpg",
    name: "Abigail Cain",
    bio: "[Abigail Cain](https://artsy.net/abigail) is an Art Market Editor at Artsy",
    twitter_handle: "abigailcain",
  },
  {
    image_url: "http://files.artsy.net/images/molly_01.jpg",
    name: "Molly Gottschalk",
    bio:
      "[Molly Gottschalk](https://artsy.net/molly) is a Features Producer at Artsy. She is a photographer and lover of cats.",
    twitter_handle: "mollygottschalk",
  },
  {
    name: "Halley Johnson",
    bio: "[Halley Johnson](https://artsy.net/molly) is an Writer Guru at Artsy. She enjoys speaking Georgian.",
    twitter_handle: "halleyjohnson",
  },
]

export const Embeds = [
  {
    mobile_height: 1300,
    height: 1000,
    url: "http://files.artsy.net/documents/1parrasch.html",
    layout: "overflow",
    type: "embed",
  },
]

export const HeroSections = [
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]

export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini: ImageSetPreviewProps["section"] = {
  type: "image_set",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetFullSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  images: Images,
}

export const ImageSetMiniSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  images: Images,
}
export const Videos = [
  {
    url: "https://www.youtube.com/watch?v=PXi7Kjlsz9A",
    caption: "<p>What motivates patrons to fund artists’ wildest dreams?</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/IB6epb5L_l0rm9btaDsY7Q%2F14183_MDP_Evening_240.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
    caption:
      "<p>2016 was a memorable year for the world, and art along with it. Powered by data culled from Artsy as well as UBS’s Planet Art app, “The Year in Art 2016” will explore how the creative community responded to the cultural shifts and tribulations this year has seen—from the destruction of Palmyra to the proliferation of Virtual Reality to the U.S. election.</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/ditbyaUgdcl6mHin07TfKA%2FMassimilianoGioni_0581.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
  },
]
