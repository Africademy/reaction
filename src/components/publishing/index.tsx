import Article from "./article"
import FeatureHeader from "./header/feature_header"
import Header from "./header/header"
import Artwork from "./sections/artwork"
import Authors from "./sections/authors"
import Caption from "./sections/caption"
import Embed from "./sections/embed"
import FullscreenViewer from "./sections/fullscreen_viewer/fullscreen_viewer"
import Image from "./sections/image"
import ImageCollection from "./sections/image_collection"
import ImageSetPreview from "./sections/imageset_preview"
import ImageSetPreviewClassic from "./sections/imageset_preview_classic"
import Text from "./sections/text"
import Video from "./sections/video"

// Icon SVGs
import ImageSet from "./icon/image_set"
import Remove from "./icon/remove"

export default {
  Article,
  Artwork,
  Authors,
  Embed,
  Caption,
  FeatureHeader,
  FullscreenViewer,
  Header,
  Image,
  ImageCollection,
  ImageSetPreview,
  ImageSetPreviewClassic,
  Text,
  Video,
  Icon: {
    ImageSet,
    Remove,
  },
}
