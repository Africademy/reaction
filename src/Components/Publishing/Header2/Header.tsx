import React from "react"
import { ArticleData } from "../Typings"
import { ClassicHeader } from "./Layouts/ClassicHeader"
import { StandardHeader } from "./Layouts/StandardHeader"

interface HeaderProps {
  article: ArticleData
  date?: string
  editLeadParagraph?: any
  editTitle?: any
  editVertical?: any
}

export const Header: React.SFC<HeaderProps> = props => {
  const { article, date, editLeadParagraph, editTitle, editVertical } = props

  switch (article.layout) {
    case "standard": {
      return (
        <StandardHeader
          article={article}
          date={date && date}
          editTitle={editTitle}
          editVertical={editVertical}
        />
      )
    }

    default: {
      return (
        <ClassicHeader
          article={article}
          date={date && date}
          editLeadParagraph={editLeadParagraph}
          editTitle={editTitle}
        />
      )
    }
  }
}
