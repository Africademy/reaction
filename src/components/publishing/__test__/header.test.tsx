import * as React from "react"
import * as renderer from "react-test-renderer"

import "jest-styled-components"
import * as _ from "lodash"

import AuthorDate from "../header/author_date"
import Header from "../header/header"
import Articles from "./fixtures/articles"
import { HeroSections } from "./fixtures/components"

jest.mock("react-sizeme", () => jest.fn(c => d => d))

describe("feature", () => {
  it("renders feature text header properly", () => {
    const article = _.extend({}, Articles[2], { hero_section: HeroSections[0] })
    const header = renderer.create(<Header article={article} />).toJSON()
    expect(header).toMatchSnapshot()
  })

  it("renders feature split header properly", () => {
    const article = _.extend({}, Articles[2], { hero_section: HeroSections[1] })
    const header = renderer.create(<Header article={article} />).toJSON()
    expect(header).toMatchSnapshot()
  })

  it("renders feature full header properly", () => {
    const article = _.extend({}, Articles[2], { hero_section: HeroSections[2] })
    const header = renderer.create(<Header article={article} />).toJSON()
    expect(header).toMatchSnapshot()
  })
})

describe("Standard Header", () => {
  it("renders standard header properly", () => {
    const header = renderer.create(<Header article={Articles[1]} />).toJSON()
    expect(header).toMatchSnapshot()
  })
})

describe("AuthorDate", () => {
  it("renders a single author", () => {
    const authors = [{ name: "Molly Gottschalk" }]
    const authorDate = renderer.create(
      <AuthorDate authors={authors} date={"2017-05-19T13:09:18.567Z"} layout={"split"} />
    )
    expect(authorDate).toMatchSnapshot()
  })

  it("renders multiple authors", () => {
    const authors = [{ name: "Molly Gottschalk" }, { name: "Kana Abe" }]
    const authorDate = renderer.create(
      <AuthorDate authors={authors} date={"2017-05-19T13:09:18.567Z"} layout={"split"} />
    )
    expect(authorDate).toMatchSnapshot()
  })
})
