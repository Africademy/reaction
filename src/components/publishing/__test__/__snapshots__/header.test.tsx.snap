// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDate renders a single author 1`] = `
<div
  className="file__AuthorDateContainer-s951nr8-0 gYwNkD"
>
  <div
    className="sc-bdVaJa bcNlZk"
  >
    By 
    Molly Gottschalk
  </div>
  <div
    className="sc-bdVaJa bcNlZk"
  >
    May 19, 2017 9:09 am
  </div>
</div>
`;

exports[`AuthorDate renders multiple authors 1`] = `
<div
  className="file__AuthorDateContainer-s951nr8-0 gYwNkD"
>
  <div
    className="sc-bdVaJa bcNlZk"
  >
    By 
    Molly Gottschalk and Kana Abe
  </div>
  <div
    className="sc-bdVaJa bcNlZk"
  >
    May 19, 2017 9:09 am
  </div>
</div>
`;

exports[`Standard Header renders standard header properly 1`] = `
<div
  className="file__StandardHeaderContainer-s13evez-0 fmppsk"
>
  <div
    className="file__Vertical-s13evez-1 jhCIcQ"
  >
    Art Market
  </div>
  <div
    className="file__StandardTitle-s1sluhjf-0 bGRsjT"
  >
    New York's Next Art District
  </div>
  <div
    className="file__AuthorDateContainer-s951nr8-0 gYwNkD"
  >
    <div
      className="sc-bdVaJa bcNlZk"
    >
      By 
      Molly Gottschalk
    </div>
    <div
      className="sc-bdVaJa bcNlZk"
    >
      May 19, 2017 9:09 am
    </div>
  </div>
</div>
`;

exports[`feature renders feature full header properly 1`] = `
<div
  className="sc-bZQynM dQyFHc"
  data-type="fullscreen"
>
  <div>
    <div
      className="sc-ifAKCX eaHLCC"
      src="https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg"
    />
    <div
      className="sc-bwzfXH gWZoXy"
    />
  </div>
  <div
    className="sc-htpNat gziOsG"
  >
    <div
      className="sc-bxivhb jzEeHb"
    >
      <div
        className="file__Vertical-lq75tn-1 eZtUXf"
      >
        Art Market
      </div>
      <div
        className="feature__title file__FeatureTitle-s1sluhjf-1 lkDrMY"
      >
        New York's Next Art District
      </div>
      <div
        className="file__SubHeader-lq75tn-5 fvCnYB"
      >
        <div
          className="feature__subheader-text file__SubHeaderText-s1sluhjf-2 jrImbk"
        >
          Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.
        </div>
        <div
          className="file__AuthorDateContainer-s951nr8-0 gYwNkD"
        >
          <div
            className="sc-bdVaJa bfrJtG"
          >
            By 
            Molly Gottschalk
          </div>
          <div
            className="sc-bdVaJa bfrJtG"
          >
            May 19, 2017 9:09 am
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`feature renders feature split header properly 1`] = `
<div
  className="sc-bZQynM dQyFHc"
  data-type="split"
>
  <div
    className="sc-htpNat gziOsG"
  >
    <div
      className="sc-bxivhb jzEeHb"
    >
      <div
        className="file__Vertical-lq75tn-1 eZtUXf"
      >
        Art Market
      </div>
      <div
        className="feature__title file__FeatureTitle-s1sluhjf-1 lkDrMY"
      >
        New York's Next Art District
      </div>
      <div
        className="sc-ifAKCX eaHLCC"
        src="https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg"
      />
      <div
        className="file__SubHeader-lq75tn-5 fvCnYB"
      >
        <div
          className="feature__subheader-text file__SubHeaderText-s1sluhjf-2 jrImbk"
        >
          Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.
        </div>
        <div
          className="file__AuthorDateContainer-s951nr8-0 gYwNkD"
        >
          <div
            className="sc-bdVaJa bcNlZk"
          >
            By 
            Molly Gottschalk
          </div>
          <div
            className="sc-bdVaJa bcNlZk"
          >
            May 19, 2017 9:09 am
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`feature renders feature text header properly 1`] = `
<div
  className="sc-bZQynM dQyFHc"
  data-type="text"
>
  <div
    className="sc-htpNat gziOsG"
  >
    <div
      className="sc-bxivhb jzEeHb"
    >
      <div
        className="file__Vertical-lq75tn-1 eZtUXf"
      >
        Art Market
      </div>
      <div
        className="feature__title file__FeatureTitle-s1sluhjf-1 lkDrMY"
      >
        New York's Next Art District
      </div>
      <div
        className="file__SubHeader-lq75tn-5 fvCnYB"
      >
        <div
          className="feature__subheader-text file__SubHeaderText-s1sluhjf-2 jrImbk"
        >
          Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.
        </div>
        <div
          className="file__AuthorDateContainer-s951nr8-0 gYwNkD"
        >
          <div
            className="sc-bdVaJa eeGSmv"
          >
            By 
            Molly Gottschalk
          </div>
          <div
            className="sc-bdVaJa eeGSmv"
          >
            May 19, 2017 9:09 am
          </div>
        </div>
      </div>
    </div>
    <img
      className="file__Image-lq75tn-3 ewYriW"
      src="https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg"
    />
  </div>
</div>
`;
